platform/ios/src/MGLMapView.mm

```diff
@@ -2496,6 +2536,22 @@ - (MGLAnnotationImage *)defaultAnnotationImage
     return annotationImage;
 }
 
+- (MGLAnnotationView *)annotationViewForAnnotation:(id<MGLAnnotation>)annotation
+{
+    MGLAnnotationView *annotationView = (MGLAnnotationView *)[self.delegate mapView:self viewForAnnotation:annotation];
```

Comment:
This cast and the assertion below are unnecessary, because `-[MGLMapViewDelegate mapView:viewForAnnotation:]` returns an MGLAnnotationView.